CMakeLists.txt

```diff
@@ -77,29 +77,37 @@ endif()
 #
 option(SAIL_BUILD_APPS "Build applications." ON)
 option(SAIL_BUILD_EXAMPLES "Build examples." ON)
+option(SAIL_BUILD_TESTS "Build tests." ON)
+option(SAIL_BUILD_BINDINGS "Build the C++ and other bindings. Set to OFF if only building C code." ON)
+option(SAIL_BUILD_SHARED_LIBS "Build shared libs. When disabled, sets SAIL_COMBINE_CODECS and SAIL_BUILD_STATIC_LIBS to ON automatically." ON)
+cmake_dependent_option(SAIL_BUILD_STATIC_LIBS "Build static libs." OFF "SAIL_BUILD_SHARED_LIBS" ON)
+cmake_dependent_option(SAIL_COMBINE_CODECS "Combine all codecs into a single library. When disabled, all codecs are implemented as \
+dynamically loaded plugins." OFF "SAIL_BUILD_SHARED_LIBS" ON)
+option(SAIL_BUILD_TYPE "Set build type: Release, Debug, or RelWithDebugInfo." "Release")
 option(SAIL_DEV "Enable developer mode. Be more strict when compiling source code, for example." OFF)
 option(SAIL_ENABLE_OPENMP "Enable OpenMP support if it's available in the compiler." ON)
+option(SAIL_ENABLE_EMBED "Enable this if Sail is a submodule for another project (like a game engine or developer tool)." OFF)
```

Comment:
This option does nothing, but prints a configuration message later. If the user embeds SAIL, he or she must be aware of how to pass CMake variables to the embedded SAIL copy. I prefer not to pollute the CMake file with too specific variables and messages.